crypto/encoding/amino/amino.go

```diff
@@ -19,6 +23,23 @@ func init() {
 	// https://github.com/tendermint/go-amino/issues/9
 	// is resolved
 	RegisterAmino(cdc)
+
+	// TODO: Have amino provide a way to go from concrete struct to route directly.
+	// Its currently a private API
+	routeTable[reflect.TypeOf(ed25519.PubKeyEd25519{})] = ed25519.PubKeyAminoRoute
+	routeTable[reflect.TypeOf(secp256k1.PubKeySecp256k1{})] = secp256k1.PubKeyAminoRoute
+	routeTable[reflect.TypeOf(&multisig.PubKeyMultisigThreshold{})] = multisig.PubKeyMultisigThresholdAminoRoute
+}
+
+// PubkeyAminoRoute returns the amino route of a pubkey
+// cdc is currently passed in, as eventually this will not be using
+// a package level codec.
+func PubkeyAminoRoute(cdc *amino.Codec, key crypto.PubKey) (string, error) {
+	route, ok := routeTable[reflect.TypeOf(key)]
+	if !ok {
+		return "", errors.New("Pubkey type not known")
```

Comment:
can we return a bool instead of error? it will simplify the code drastically

```
return routeTable[reflect.TypeOf(key)]
```

Reply:
Sure!

Reply:
Ff